feat!: Implement query cancellation in react-query generated components

needim opened the pull request:
This PR contains abort signal support for reactQuery.

But I had some trouble with my local setup. Tests didn't run correctly at first. I cleared the jest cache but nothing was resolved. I re-forked the repository but had the same issue again. So I removed package-lock and npm install again to solve the issue. (Maybe previous version-bump PR broked something, I am not sure)

But the second problem is related to building. The build doesn't work right now on my local setup. Maybe it is related to version-bump of Lerna again but not sure.

Other notes: 
- tests are running via Yarn In Github actions.
- In tests, we are using the `got` package but it is not in the devDependencies.

I am confused but opened PR to see if tests are running in Github actions.

needim commented:
Looks like the tests are passed. But I'm still not sure about this PR atm.

fabien0102 commented:
This looks good, amazing PR! 🎉 I will have a look later, I'm curious about this `got` 🤔

needim commented:
that `got` thing happened to me about last month about `redis` we were using `handy-redis` package and we were able to import `redis` for like more than one year without any problem. (maybe because it is dependent on redis) after some version bumps we got some errors on CI about redis (maybe packages changed their dependency from devDep to peerDep I am still not sure)

maybe this is related to my local but can you try to repo;

1. clone the repository in a new folder
2. run npm i 
3. run npm test
4. run lerna run build

needim commented:
and sorry for ruining your Friday night :)

needim commented:
Ok, I fixed the build issue. It now correctly builds all the packages. 

100% sure I broke it with the previous version-bump PR (lack of Lerna knowledge). And I added `got` as devDependency in the main package.json. 

Also, I removed package-lock files from packages. Looks like lerna/workspace handles all with one single lock.

needim commented:
closes: #94

selcukfatihsevinc commented:
any updates?

fabien0102 commented on plugins/typescript/src/generators/generateFetchers.test.ts:
```diff
@@ -140,7 +140,7 @@ describe("generateFetchers", () => {
        * Get all the pets
        */
-      export const listPets = () => fetch<ListPetsResponse, ListPetsError, undefined, {}, {}, {}>({ url: \\"/pets\\", method: \\"get\\" });
+      export const listPets = () => fetch<ListPetsResponse, ListPetsError, undefined, {}, {}, {}>({ url: \\"/pets\\", method: \\"get\\", signal });
```
The signal argument is missing here

fabien0102 commented:
I tried locally, everything is following, @needim do you to have a quick look to see if I missed something?

needim commented:
It should be

> I tried locally, everything is following, @needim do you to have a quick look to see if I missed something?

Your changes are super cool! Looks all right to me! Moving signal from context to fetchers 🎯👍 🎉

_This library isn't only about generating reactQuery components, it can only generate fetchers too. I should write this to my head_ ☺️
